lib/web_ui/lib/src/ui/text.dart

```diff
@@ -1069,42 +1070,42 @@ class TextRange {
   /// The index of the first character in the range.
   ///
   /// If [start] and [end] are both -1, the text range is empty.
-  final int start;
+  final int/*!*/ start;
 
   /// The next index after the characters in this range.
   ///
   /// If [start] and [end] are both -1, the text range is empty.
-  final int end;
+  final int/*!*/ end;
 
   /// Whether this range represents a valid position in the text.
-  bool get isValid => start >= 0 && end >= 0;
+  bool/*!*/ get isValid => start >= 0 && end >= 0;
 
   /// Whether this range is empty (but still potentially placed inside the text).
-  bool get isCollapsed => start == end;
+  bool/*!*/ get isCollapsed => start == end;
 
   /// Whether the start of this range precedes the end.
-  bool get isNormalized => end >= start;
+  bool/*!*/ get isNormalized => end >= start;
 
   /// The text before this range.
-  String textBefore(String text) {
+  String/*!*/ textBefore(String text) {
```

Comment:
Here and below do the parameters need to be annotated?

Reply:
Done.